bot/cogs/candy_collection.py

```diff
@@ -0,0 +1,210 @@
+             (higher probability)"""
+
+        if random.randint(1, self.ADD_SKULL_EXISTING_REACTION_CHANCE) == 1:
+            d = {"reaction": '\N{SKULL}', "msg_id": message.id, "won": False}
+            self.msg_reacted.append(d)
+            return await message.add_reaction('\N{SKULL}')
+
+        if random.randint(1, self.ADD_CANDY_EXISTING_REACTION_CHANCE) == 1:
+            d = {"reaction": '\N{CANDY}', "msg_id": message.id, "won": False}
+            self.msg_reacted.append(d)
+            return await message.add_reaction('\N{CANDY}')
+
+    async def ten_recent_msg(self):
+        """Get the last 10 messages sent in the channel"""
+        ten_recent = []
+        recent_msg = max(message.id for message
+                         in self.bot._connection._messages
+                         if message.channel.id == self.HACKTOBER_CHANNEL_ID)
+
+        channel = await self.hacktober_channel()
+        ten_recent.append(recent_msg.id)
+
+        for i in range(9):
+            o = discord.Object(id=recent_msg.id + i)
+            msg = await channel.history(limit=1, before=o).next()
+            ten_recent.append(msg.id)
+
+        return ten_recent
+
+    async def get_message(self, msg_id):
+        """Get the message from it's ID."""
+
+        try:
+            o = discord.Object(id=msg_id + 1)
+            # Use history rather than get_message due to
+            #         poor ratelimit (50/1s vs 1/1s)
+            msg = await self.hacktober_channel.history(limit=1, before=o).next()
+
+            if msg.id != msg_id:
+                return None
+
+            return msg
+
+        except Exception:
+            return None
+
+    async def hacktober_channel(self):
+        """Get #hacktoberbot channel from it's id"""
+        return self.bot.get_channel(id=self.HACKTOBER_CHANNEL_ID)
+
+    async def remove_reactions(self, reaction):
+        """Remove all candy/skull reactions"""
+
+        try:
+            async for user in reaction.users():
+                await reaction.message.remove_reaction(reaction.emoji, user)
+
+        except discord.HTTPException:
+            pass
+
+    async def send_spook_msg(self, author, channel, candies):
+        """Send a spooky message"""
+        e = discord.Embed(colour=author.colour)
+        e.set_author(name="Ghosts and Ghouls and Jack o' lanterns at night; "
+                          f"I took {candies} candies and quickly took flight.")
+        await channel.send(embed=e)
+
+    def save_to_json(self):
+        """Save json to the file."""
+        with open(json_location, 'w') as outfile:
+            json.dump(self.candy_json, outfile)
+
+    @commands.command()
+    async def candy(self, ctx):
+        """Get the candy leaderboard and save to json when this is called"""
+
+        # use run_in_executor to prevent blocking
+        thing = functools.partial(self.save_to_json)
+        save = await self.bot.loop.run_in_executor(None, thing)
```

Comment:
From the linter:
`bot/cogs/candy_collection.py:176:9: F841 local variable 'save' is assigned to but never used`